akka-remote/src/main/scala/akka/remote/security/provider/AESCounterBuiltinRNG.scala

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright (C) 2016-2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.remote.security.provider
+
+import java.security.Key
+import java.util.Random
+import java.util.concurrent.locks.ReentrantLock
+import javax.crypto.Cipher
+
+/**
+ * INTERNAL API
+ * This class is a Scala implementation of AESCounterRNG algorithm
+ * patterned after org.uncommons.maths.random by Daniel Dyer (Apache License 2.0)
+ *
+ * Non-linear random number generator based on the AES block cipher in counter mode.
+ * Uses the seed as a key to encrypt a 128-bit counter using AES(Rijndael).
+ *
+ * Keys larger than 128-bit for the AES cipher require
+ * the inconvenience of installing the unlimited strength cryptography policy
+ * files for the Java platform.  Larger keys may be used (192 or 256 bits) but if the
+ * cryptography policy files are not installed, a
+ * java.security.GeneralSecurityException will be thrown.
+ *
+ * NOTE: THIS CLASS IS NOT SERIALIZABLE
+ */
+
+class AESCounterBuiltinRNG(val seed: Array[Byte]) extends Random {
+  private val COUNTER_SIZE_BYTES = 16
+  private val BITWISE_BYTE_TO_INT = 0x000000FF
+
+  // mutable state below, concurrent accesses need lock
+  private val lock = new ReentrantLock
+  private val counter: Array[Byte] = new Array[Byte](COUNTER_SIZE_BYTES)
+  private var index: Int = 0
+  private var currentBlock: Array[Byte] = null
+
+  private val cipher = Cipher.getInstance("AES/ECB/NoPadding")
```

Comment:
I have added `AESCounterBuiltinCTRRNG` that is using `AES/CTR/NoPadding`. To allow verification I have added custom iv with value of 1 but, obviously, this can be removed before merge.
Works great, only issue being that it is even slower, as the zero block needs now to be XORed with the encrypted counter. So the performance looks like that:
```
[info] Benchmark                          Mode  Cnt    Score    Error  Units
[info] RandomReplacementBenchmark.runCtr    ss   15  299.485 ± 23.466  ms/op
[info] RandomReplacementBenchmark.runNew    ss   15  238.772 ± 12.127  ms/op
[info] RandomReplacementBenchmark.runOld    ss   15  215.453 ± 24.637  ms/op

[info] RandomReplacementBenchmark.runCtr    ss   15  283.690 ± 17.258  ms/op
[info] RandomReplacementBenchmark.runNew    ss   15  251.412 ± 25.066  ms/op
[info] RandomReplacementBenchmark.runOld    ss   15  231.133 ± 29.139  ms/op

[info] RandomReplacementBenchmark.runCtr    ss   15  314.895 ± 45.711  ms/op
[info] RandomReplacementBenchmark.runNew    ss   15  250.495 ± 15.762  ms/op
[info] RandomReplacementBenchmark.runOld    ss   15  215.201 ± 23.651  ms/op
```
which means that the CTR algo is **1.5x** slower than the original one. If that is fine, great, I will happily replace that as hand-built counter is also hurting my eyes.
Of course, aye to reseeding. This is missing from the original implementation, and, for realistic messaging volumes, badly needed. I will add this as soon as we settle the algorithm.